extensions/wikia/ThemeDesigner/js/ThemeDesigner.js

```diff
@@ -1,5 +1,6 @@
-/* global $ */
-(function (window) {
+require(
+	['jquery', 'mw', 'wikia.nirvana', 'wikia.fluidlayout', 'wikia.window'],
+	function($, mw, nirvana, fluidlayout, window) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This function has 1072 lines, which is greater than the 100 lines authorized. Split it into smaller functions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS138)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has a complexity of 171 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)